test/integration/change-streams/change_stream.test.ts

```diff
@@ -2138,6 +2138,185 @@ describe('ChangeStream resumability', function () {
         );
       });
     });
+
```

Comment:
Two other important tests:

1. we should demonstrate that we cannot use Symbol.asyncIterator in conjunction with the event emitter based iteration. So the following should throw an error (and I believe it already does):

```typescript
const cs = collection.watch([]);
cs.on('change', change => { .... });

for await (const change of cs) { // this should throw

}
```

2. we should demonstrate that we can use Symbol.asyncIterator in conjunction with the raw iterator based api:
```typescript
const cs = collection.watch([]);

await cs.next();  // iterate a bit using the iterator API

for await (const change of cs) { // this should not throw

}
```